### ISSUE TITLE
Local Jitsi isn't used consistently

### ISSUE THREAD

**Issue by jgoerzen** [spantaleev/matrix-docker-ansible-deploy#450]
Hi,

Thanks for making this!

So I enabled the local Jitsi.  Here is what I've observed:

1) If I click the video icon in a room on my local HS while logged into my local riot, it creates a Jitsi room locally.

2) When connected to the room set up as with step 1 from a federated Riot, the Jitsi never shows up when embedded, but can be popped out and works the.

3) When connected to the room set up as with step 1 from the riot Android client, it connects to the wrong server; it goes to a room of the same name on jitsi.riot.im.

4) When using the local Dimension selector to add a widget, it defaults to jitsi.riot.im.

**Comment by spantaleev** [comment]
1. Great, works as expected!

2. It currently only uses your own server if riot-web's configuration points there. We set up your own riot-web to do that, but other riot-web installations have their own configuration and won't respect our settings. The change I made in 7035af87d8d96c9 is supposed to improve the situation, but it will only work when you're logging into your own account from another riot-web, which is v1.5.16 (not yet released). So.. expect things to improve soon.

3. As the [riot-web Jitsi documentation page](https://github.com/vector-im/riot-web/blob/6e15684a044fe536e094219992260dc00db3dfc0/docs/jitsi.md#mobile-app-support) says:

> Currently the Riot mobile apps do not support custom Jitsi servers and will instead use the default jitsi.riot.im server. When users on the mobile apps join the call, they will be joining a different conference which has the same name, but not the same participants. This is a known bug and which needs to be fixed.

4. Possibly.. The playbok does not configure Dimension to point it to our self-hosted Jitsi server. I'm not sure it's possible to pre-configure it through the Dimension configuration file. If it is, we should do it and make this a no-brainer.

**Comment by jgoerzen** [comment]
Thanks!  Based on this, I went and re-disabled Jitsi.  Unfortunately, clicking on the camera icon while connected to the local riot causes it to still attempt to use the local Jitsi rather than the riot.im one.  

I believe issue 2 is related to something around CORS restrictions, and perhaps I should report it elsewhere.  If I use the local Dimension to create a Jitsi widget, it will use riot.im but a federated Jitsi wants to talk to https://dimension.DOMAIN.  It works when popped out from a federated riot, just not when on the same page as the channel.

**Comment by spantaleev** [comment]
> Thanks! Based on this, I went and re-disabled Jitsi. Unfortunately, clicking on the camera icon while connected to the local riot causes it to still attempt to use the local Jitsi rather than the riot.im one.

It means you probably haven't rebuilt and restarted all services properly (`ansible-playbook ... --tags=setup-all,start`). Or you haven't fully reloaded riot-web, so it hasn't fetched its updated configuration (`https://riot.DOMAIN/config.json`), which should stop mentioning the local Jitsi anymore, if all is well.

**Comment by jgoerzen** [comment]
I had actually re-run `ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start` and verified that the jitsi docker containers were gone after it ran.  The references in .well-known/matrix/client were also removed and are properly being served up via proxying.  I have also rebooted the VM it's running in, and it's still there.

Thanks, by the way, for all your help and for putting this excellent system together!
